src/GtirbZeroBuilder.cpp

```diff
@@ -114,9 +116,51 @@ void buildSections(gtirb::Module &module, std::shared_ptr<BinaryReader> binary,
             module.addSection(section);
             sectionProperties[section->getUUID()] =
                 std::make_tuple(binSection.type, binSection.flags);
+            allSections[section->getUUID()] =
+                std::make_tuple(binSection.type, binSection.flags);
+        } else {
+            //Check if it is a DWARF section
+            static std::vector<std::string> sectionNames{
+                ".debug_abbrev",
+                ".debug_aranges",
+                ".debug_frame",
+                ".debug_info",
+                ".debug_line",
+                ".debug_loc",
+                ".debug_macinfo",
+                ".debug_types",
+                ".debug_str",
+                ".debug_ranges",
+                ".debug_pubnames",
+                ".debug_pubtypes"
+            };
+
+            auto gtirbName = binSection.name;
+            gtirb::Section *section = gtirb::Section::Create(
+                context, binSection.name, gtirb::Addr(binSection.address),
+                binSection.size);
+            for(std::string sName : sectionNames)
+            {
+                if(gtirbName == sName)
+                {
+                    //Detected the debug section
+                    //Debug section isn't allocatable
+                    module.addSection(section);
+                    dwarfSections[section->getUUID()] =
+                        std::make_tuple(binSection.type, binSection.flags);;
+
+                    break;
+                }
+            }
+            allSections[section->getUUID()] =
+                std::make_tuple(binSection.type, binSection.flags);
         }
+        //Logs all sections in a separate aux component
+
     }
     module.addAuxData("elfSectionProperties", std::move(sectionProperties));
+    module.addAuxData("dwarfSections", std::move(dwarfSections));
```

Comment:
These have been changed to templates using types defined in AuxDataSchema

Reply:
Ah! Just checked out latest master, just need to place a type argument. I'll get this modification done asap.